.github/workflows/linting.yaml

```diff
@@ -2,20 +2,31 @@ name: Golang Lint
 on: [pull_request]
 
 jobs:
-  golangci:
-    name: lint
-    runs-on: ubuntu-20.04
+  golangci-lint:
+    name: golangci lint
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Setup go
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@v5
         with:
-          go-version: '1.22'
-      - name: golangci-lint go
-        uses: golangci/golangci-lint-action@v3
+          go-version-file: 'go.work'
+          cache: false
+      - name: lint go
+        uses: golangci/golangci-lint-action@v4
         with:
           # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
-          version: v1.54
+          version: v1.57
           working-directory: opensearch-operator
           args: --timeout=6m --skip-dirs="(^|/)responses($|/)" -v
-
+ 
+  helm-lint:
```

Comment:
NIT. Hey @salyh can we have this `helm-lint` as a seperate workflow and trigger only when there is a change in files under `charts/` folder, this will make CI faster and avoid this check when there is no change with helm charts.
Thanks

Reply:
Done